### PR TITLE
docs: add API-using examples to module and layout guides

### PR DESCRIPTION
**Pull request by alextremblay**
Fleshing out an example provided in https://github.com/directus/directus/discussions/4878 by @rijkvanzanten, I've added more detailed examples to the layout and module guides showing people how to access and work with the API inside of their own custom extensions

### CODE REVIEW

File: docs/guides/layouts.md [hunk 1]
```diff
@@ -63,6 +63,54 @@ The props you can use in an layout are:
 - `filters` (sync) - The user's currently active filters.
 - `search-query` (sync) - The user's current search query.
 
+
+#### Accessing the API from within your extension
+The Directus App's Vue app instance provides a field called `system`, which can be injected into Vue components 
+using [Vue's inject framework](https://v3.vuejs.org/guide/component-provide-inject.html). This `system` field contains functions to access Vuex stores, and more importantly, contains a property called `api`, which is an authenticated Axios instance. Here's an example of how to use it:
```
Comment (rijkvanzanten) [inline docs/guides/layouts.md:69]:
```suggestion
using [Vue's inject framework](https://v3.vuejs.org/guide/component-provide-inject.html). This `system` field contains functions to access [Pinia](https://pinia.esm.dev) stores, and more importantly, contains a property called `api`, which is an authenticated Axios instance. Here's an example of how to use it: 
```

File: docs/guides/layouts.md [hunk 2]
```diff
@@ -63,6 +63,54 @@ The props you can use in an layout are:
...
+    // log the system field so you can see what attributes are available under it
+    // remove this line when you're done.
+    console.log(this.system)
+
+    // Get a list of all available collections to use with this module
+    this.system.api.get(`/items/${this.collection}`)
+      .then((res) => {this.items = res});
+  }
+}
+</script>
```
Comment (rijkvanzanten) [inline docs/guides/layouts.md:103]:
```suggestion
<template>
  <div>
    <div>Collection: {{ collection }}</div>
    <v-list>
      <v-list-item v-for="item in items" v-bind:key="item.id">
        {{item}}
      </v-list-item>
    </v-list>
    <v-button v-on:click="logToConsole">CLog items to console</v-button>
  </div>
</template>
<script>
  export default {
    data() {
      return {
        items: null,
      };
    },
    methods: {
      logToConsole: function () {
        console.log(this.items);
      },
    },
    inject: ["system"],
    mounted() {
      // log the system field so you can see what attributes are available under it
      // remove this line when you're done.
      console.log(this.system);
      // Get a list of all available collections to use with this module
      this.system.api.get(`/items/${this.collection}`).then((res) => {
        this.items = res;
      });
    },
  };
</script>
```

File: docs/guides/modules.md [hunk 3]
```diff
@@ -57,6 +57,54 @@ export default {};
...
+    // remove this line when you're done.
+    console.log(this.system)
+
+    // Get a list of all available collections to use with this module
+    this.system.api.get('/collections?limit=-1')
+      .then((res) => {this.collections = res.data.data});
+  }
+}
+</script>
+```
```
Comment (rijkvanzanten) [inline docs/guides/modules.md:98]:
```suggestion
```vue
<template>
  <private-view title="Example Collection List">
    <v-list>
      <v-list-item v-for="col in collections" v-bind:key="col.collection">
        {{col.collection}}
      </v-list-item>
    </v-list>
    <v-button v-on:click="logToConsole">Log collections to console</v-button>
  </private-view>
</template>

<script>
  export default {
    data() {
      return {
        collections: null,
      };
    },
    methods: {
      logToConsole: function () {
        console.log(this.collections);
      },
    },
    inject: ["system"],
    mounted() {
      // log the system field so you can see what attributes are available under it
      // remove this line when you're done.
      console.log(this.system);

      // Get a list of all available collections to use with this module
      this.system.api.get("/collections?limit=-1").then((res) => {
        this.collections = res.data.data;
      });
    },
  };
</script>
```
```

**Review by rijkvanzanten — COMMENTED** [review]
Changes in examples are just consistent formatting

File: docs/guides/modules.md [hunk 4]
```diff
@@ -57,6 +57,56 @@ export default {};
...
+      // remove this line when you're done.
+      console.log(this.system);
+
+      // Get a list of all available collections to use with this module
+      this.system.api.get("/collections?limit=-1").then((res) => {
+        this.collections = res.data.data;
+      });
+    },
+  };
+</script>
```
Comment (rijkvanzanten) [inline docs/guides/modules.md:100]:
```suggestion
</script>
```
```

**Review by rijkvanzanten — APPROVED** [review]

**Comment by alextremblay** [comment]
Looks great. Thanks!

**Pull request merged**